invenio_records_resources/records/systemfields/files.py

```diff
@@ -202,6 +214,34 @@ def __getitem__(self, key):
                 return value
         raise KeyError(f'No file with key "{key}"')
 
+    def _parse_set_value(self, value):
+        obj, stream, data = None, None, None
+        # TODO: Raise appropriate exceptions instead of asserting
+        obj_or_stream = None
+        if isinstance(value, (tuple, list)):
+            assert len(value) == 2
+            obj_or_stream, data = value
+            assert isinstance(data, dict)
+        elif isinstance(value, dict):
+            data = value
+        elif isinstance(value, ObjectVersion):
+            obj = value
+        elif hasattr(value, 'read'):
+            stream = value
+        else:
+            raise Exception(f"Invalid set value: {value}")
+
+        if obj_or_stream:
+            if isinstance(obj_or_stream, ObjectVersion):
+                obj = obj_or_stream
+            elif hasattr(obj_or_stream, 'read'):
```

Comment:
Comment: You cannot distinguish also here like above if it's a stream by its Class? Would feel safer if so.

Reply:
It seems to be a bit more difficult to figure this out: https://stackoverflow.com/questions/1661262/check-if-object-is-file-like-in-python